core/src/player.rs

```diff
@@ -1539,8 +1539,8 @@ impl Player {
     pub fn run_actions(context: &mut UpdateContext<'_, '_, '_>) {
         // Note that actions can queue further actions, so a while loop is necessary here.
         while let Some(action) = context.action_queue.pop_action() {
-            // We don't run frame actions if the clip was removed after it queued the action.
-            if !action.is_unload && action.clip.removed() {
+            // We don't run frame actions if the clip was removed (or scheduled to be removed) after it queued the action.
+            if !action.is_unload && (action.clip.removed() || action.clip.pending_removal()) {
```

Comment:
I think the `is_unload` flag can be nicely removed now, as this flag was making up for the missing unload behavior, and this check can be handled via  the `pending_removal` flag now. Might need to be:
`if action.clip.removed() && !action.clip.pending_removal()`
(i.e. allow the clip to still execute code if it's being delay-removed).

Reply:
I'm not sure this can be removed, I gave it a try but it breaks unload events on non-timeline clips, as well as some enterFrame events. I could fix (at least the first one) by treating them as pending but that will likely have issues due to the other effects of the pending flag (such as changing their position in the depth list).